libpod/runtime_ctr.go

```diff
@@ -246,6 +249,40 @@ func (r *Runtime) newContainer(ctx context.Context, rSpec *spec.Spec, options ..
 }
 
 func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Container, retErr error) {
+	if ctr.IsDefaultInfra() || ctr.IsService() {
+		tmpDir, err := r.TmpDir()
+		if err != nil {
+			return nil, fmt.Errorf("getting runtime temporary directory: %w", err)
+		}
+		tmpDir = filepath.Join(tmpDir, "infra-containers", ctr.ID())
+		err = os.MkdirAll(tmpDir, 0777)
+		if err != nil {
+			return nil, fmt.Errorf("creating infra container temporary directory: %w", err)
+		}
+		// Also look into the path as some distributions install catatonit in
+		// /usr/bin.
+		rtc, err := r.GetConfigNoCopy()
+		if err != nil {
+			return nil, fmt.Errorf("getting Runtime configuration: %w", err)
+		}
+		catatonitPath, err := rtc.FindInitBinary()
```

Comment:
when you are in libpod you can directly access `r.config`, just be sure to not overwrite options.